components/camel-olingo4/camel-olingo4-component/src/test/java/org/apache/camel/component/olingo4/Olingo4ComponentTest.java

```diff
@@ -402,6 +409,49 @@ public void testProducerReadFilterAlreadySeen() throws Exception {
         }
     }
 
+    /**
+     * Read entity set of the People object and split the results
+     * into individual messages
+     */
+    @Test
+    public void testConsumerReadSplitResults() throws Exception {
+        final Map<String, Object> headers = new HashMap<>();
+        String endpoint = "olingo4://read/People?consumer.splitResult=true";
+        int expectedEntities = 20;
+        final ClientEntitySet entities = (ClientEntitySet)requestBodyAndHeaders(endpoint, null, headers);
+        assertNotNull(entities);
+        assertEquals(expectedEntities, entities.getEntities().size());
+
+        int expectedMsgCount = 3;
+        MockEndpoint mockEndpoint = getMockEndpoint("mock:consumer-splitresult");
```

Comment:
You should set your mock expectations before you send data into Camel, so move these lines up in the beginning of the test method, then send data, and then assert on the mock. Do the same in the other unit tests.

Reply:
Done.